server/models/User.coffee

```diff
@@ -1,77 +1,102 @@
 {Schema} = require 'mongoose'
+formatFields = require 'mongoose-format-fields'
 
-User = new Schema
+Model = new Schema
   # fb fields
   fbid:
+    grants: ['admin']
     type: String
     required: true
     index:
       unique: true
 
   name:
+    grants: ['admin']
     type: String
     required: true
 
   first_name:
+    grants: 'public'
     type: String
     required: true
 
   last_name:
+    grants: ['admin']
     type: String
     required: true
 
   token:
+    grants: ['admin']
     type: String
     required: true
 
-  link:
-    type: String
-
-  username:
-    type: String
-    index:
-      unique: true
-      sparse: true
-
   gender:
+    grants: 'public'
     type: String
+    lowercase: true
 
   timezone:
+    grants: ['admin']
     type: Number
 
   locale:
+    grants: ['admin']
     type: String
 
   verified:
+    grants: ['admin']
     type: Boolean
 
   location:
+    grants: 'public'
     type: String
     default: 'Earth'
 
+  bio:
+    grants: 'public'
+    type: String
+
   created:
+    grants: ['admin']
     type: Date
     default: Date.now
 
   lastModified:
+    grants: ['admin']
     type: Date
     default: Date.now
 
   # custom fields here
-  points:
-    type: Number
-    default: 0
-    min: 0
+  ready:
+    grants: ['admin']
+    type: Boolean
+    default: false
 
-User.virtual('prettyName').get ->
-  return @username or @fbid
+  nextReservation:
+    grants: ['admin']
+    type: Schema.Types.ObjectId
+    ref: 'Reservation'
+
+  phone:
+    grants: 'public'
+    type: String
+    validate: /^\d{10}$/
+    unique: true
 
-User.set 'toJSON', {getters:true, virtuals:true}
-User.set 'toObject', {getters:true, virtuals:true}
-User.set 'strict', true
+Model.virtual('image').get ->
+  return "http://graph.facebook.com/#{@fbid}/picture?height=800&type=normal&width=800"
 
-User.pre 'save', (next) ->
+Model.set 'id_grants', 'public'
```

Comment:
comment also needs these two lines (89 and 90) to tell the format fields plugin not to hide the id